go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb

```diff
@@ -90,32 +92,29 @@ def update_files # rubocop:disable Metrics/AbcSize
             run_go_vendor
             run_go_mod_tidy
 
-            # At this point, the go.mod returned from run_go_get contains the
-            # correct set of modules, but running `go get` can change the file
-            # in undesirable ways (such as injecting the current Go version),
-            # so we need to update the original go.mod with the updated set of
-            # requirements rather than using the regenerated file directly
-            original_reqs = original_manifest["Require"] || []
-            updated_reqs = parse_manifest["Require"] || []
-
-            original_paths = original_reqs.map { |r| r["Path"] }
-            updated_paths = updated_reqs.map { |r| r["Path"] }
-            req_paths_to_remove = original_paths - updated_paths
-
-            # Put back the original content before we replace just the updated
-            # dependencies.
-            write_go_mod(original_go_mod)
-
-            remove_requirements(req_paths_to_remove)
-            deps = updated_reqs.map { |r| requirement_to_dependency_obj(r) }
-            update_go_mod(deps)
-
             # put the old replace directives back again
             substitute_all(substitutions.invert)
 
             updated_go_sum = original_go_sum ? File.read("go.sum") : nil
             updated_go_mod = File.read("go.mod")
 
+            # running "go get" may inject the current go version, remove it
+            original_go_version = original_go_mod.match(GO_MOD_VERSION)
+            updated_go_version = updated_go_mod.match(GO_MOD_VERSION)
+            if original_go_version != updated_go_version
+              go_mod_lines = updated_go_mod.lines
+              go_mod_lines.each_with_index do |line, i|
+                next unless line&.match?(GO_MOD_VERSION)
+
+                # replace with the original version
+                go_mod_lines[i] = original_go_version
```

Comment:
Aside: I'm pretty sure this also only matters if `original_go_version` is nil.

What dependabot does with the `go` command won't ever change the required go version(?), only add it if it was missing.
I wrote it defensively just in case, to match the previous behaviour.